libpgcommon/shared_gserialized.c

```diff
@@ -0,0 +1,98 @@
+/**********************************************************************
+ *
+ * PostGIS - Spatial Types for PostgreSQL
+ * http://postgis.net
+ *
+ * PostGIS is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * PostGIS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PostGIS.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **********************************************************************
+ *
+ * Copyright 2020 Raúl Marín
+ *
+ **********************************************************************/
+
+#include "lwgeom_cache.h"
+
+SHARED_GSERIALIZED *
+shared_gserialized_new_nocache(Datum d)
```

Comment:
I don't have my head entirely around the lifecycle of this, but is there a place for a `void shared_gserialized_new_nocache(Datum d, SHARED_GSERIALIZED *s)` and passing a statically allocated `SHARED_GSERIALIZED` in? Probably the world's tiniest optimization, given that the next step is copying in a whole `GSERIALIZED`, but it saves the `palloc`

Reply:
This is only used in `ToastCacheGetGeometry` which will return the pointer (which is good since we want to return a pointer to the local context when not caching, and a pointer to the cache context when caching) so the palloc will be needed either here or there.